### PR TITLE
Remove Gatsby alias and use Drupal alias instead

### PR DESCRIPTION
**Pull request by wsadmin**
### Frontend

- Remove alias-generating functions in `gatsby-node.js` (see e660028 for details)
- Remove mapping from `gatsby-config.js`
- Update `index.js` to use `node.path.alias` in `graphQL` query
- Replace tabs with spaces in updated files

### Backend

- Add composer patching capability as per https://duvien.com/blog/how-apply-patch-file-composer-based-drupal-89
- Update `composer.json` to add [paragraph revision fix patch](https://www.drupal.org/files/issues/2020-06-19/nested-paragraph-revision-jsonapi-workflow.patch)
- Update URL alias pattern for `Program` content type
- Add Gatsby Preview button to `Basic Page` and `Program` content types
- Add Linkit module and CHUG Linkit profile
- Update `Basic HTML` and `Full HTML` text formats to enable and use Linkit for internal links
- Update Gatsby settings to send `URL alias` to Gatsby Cloud as well

### Test plan

1. Go to multidev at https://gcloud-chug.pantheonsite.io/
2. Create a new basic page. The `Gatsby Preview` button should not be visible yet
3. Add a `general text` widget with text and save the new page as published
4. Go to the [gcloud site on Gatsby Cloud](https://www.gatsbyjs.com/dashboard/2e86c058-a370-4898-ae51-9698cd178e8d/sites/ac44984b-c63e-411d-8f1d-e96fb9305865/cmsPreview) and observe if a new Preview build occurred 
5. Go back to the page you created and update the existing `general text` widget
6. Save as draft
7. Click the Gatsby Preview button (which should now be visible)
8. The Preview page should open in a new window. Once the preview build completes, you should see your updated text
9. Repeat the above steps for a new program page
10. Repeat the above steps for a new basic page, but leave it in Draft mode throughout. It should still be previewable once saved and each successive save should trigger a new preview build that shows updated paragraph content.

### CODE REVIEW

**Comment by brharp** [comment]
At step 8 in the test plan, Gatsby tells me it is generating a new preview as expected, but when I click to view the new preview it displays the old (published) text not the new (draft) text.

**Comment by wsadmin** [comment]
> At step 8 in the test plan, Gatsby tells me it is generating a new preview as expected, but when I click to view the new preview it displays the old (published) text not the new (draft) text.

Right, this is a bug in Drupal core that the patch I applied doesn't quite address. It has to do with Drupal and the Content Moderation module not playing nice with JSON when it comes to paragraph revisions. If you'd updated a non-paragraph aspect of the basic page, e.g. uploaded a feature image, it would have shown up correctly in the preview.

Here's a revised test plan for Step 8 - see if this works:

1. Go to https://gcloud-chug.pantheonsite.io/admin/content
2. Check the box next to your test page
3. In the `Action` dropdown menu choose `Unpublish current revision`
4. Click OK
5. Make another edit to the page and save as draft
6. This time, it should show up correctly in the Preview.

So to sum up, if you keep content in "draft" mode, Gatsby Preview will correctly build every update even if it's to a paragraph. If you update a paragraph in published content and save as draft, Gatsby Preview fails to register the change because Drupal core and Content moderation don't let it know there's been a change. The workaround is to unpublish the latest revision as per the steps above.

**Comment by brharp** [comment]
The workaround worked. If I unpublish the current revision, I can see draft changes in the preview. It sort of limits the utility of the preview though---once a page is published it's not always possible to unpublish the page in order to make revisions.

**Comment by wsadmin** [comment]
> The workaround worked. If I unpublish the current revision, I can see draft changes in the preview. It sort of limits the utility of the preview though---once a page is published it's not always possible to unpublish the page in order to make revisions.

It seems that unpublishing the page on Drupal doesn't cause a new production build - I still see the original content on https://gcloud67506.gatsbyjs.io/test-mbh

So perhaps this is a limitation we can live with, given there's no other way around it until Drupal fixes the root cause in core.

**Review by snschau — APPROVED** [review]
Based on our discussions in Huddle
